specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualMachines.json

```diff
@@ -0,0 +1,801 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualmachines under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/virtualmachinesListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "virtualmachinesProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "hardwareProfile": {
+          "type": "object",
+          "properties": {
+            "vmSize": {
+              "type": "string"
+            },
+            "processors": {
+              "type": "integer",
+              "format": "int32",
+              "description": "number of processors for the virtual machine"
+            },
+            "memoryGB": {
+              "type": "integer",
+              "format": "int32",
+              "description": "RAM in gb for the virtual machine"
+            },
+            "dynamicMemoryConfig": {
+              "type": "object",
+              "properties": {
+                "maximumMemoryGB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "minimumMemoryGB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "targetMemoryBuffer": {
+                  "type": "integer",
+                  "format": "int32"
+                }
+              }
+            }
+          },
+          "description": "HardwareProfile - Specifies the hardware settings for the virtual machine."
+        },
+        "networkProfile": {
+          "type": "object",
+          "properties": {
+            "networkInterfaces": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "id": {
+                    "type": "string",
+                    "description": "ID - Resource Id"
+                  }
+                }
+              },
+              "description": "NetworkInterfaces - list of network interfaces to be attached to the virtual machine"
+            }
+          },
+          "description": "NetworkProfile - describes the network configuration the virtual machine"
+        },
+        "osProfile": {
+          "type": "object",
+          "properties": {
+            "adminPassword": {
```

Comment:
>adminPassword [](http://example.com/codeflow?start=13&length=13)

Secrets are not allowed in GET payloads. If it needs to be retrievable, it must be via a separate POST action.

Reply:
Good point, we should remove this from the GET operation.

Reply:
@Arkanayan ^